rust/src/write.rs

```diff
@@ -406,7 +428,12 @@ impl<'a, W: Write + Seek> Writer<'a, W> {
             WriteMode::Raw(w) => {
                 // It's chunkin time.
                 self.stats.chunk_count += 1;
-                WriteMode::Chunk(ChunkWriter::new(w, self.options.compression)?)
+                WriteMode::Chunk(ChunkWriter::new(
+                    w,
+                    self.options.compression,
+                    #[cfg(feature = "zstd")]
+                    self.options.zstd_worker_threads,
```

Comment:
Is there a nice way to make this `compression_options` or some way to allow for configuring options that might come up for other compression schemes? Maybe this is the way this is done in rust and we would add more feature checks while adding more arguments to the function? To me it feels unnatural if I use this library with the `zstd` feature and the `lz4` feature (for example) and that means my list of arguments grows unbounded here?

Reply:
Yes, I agree that it is unnatural and I was mostly following the existing paradigm. If we're not too concerned about API breakage (it's not `1.0` after all) I'll put up a PR for how I think the configuration should ideally look like.

Reply:
I personally would like to see that to help us shape the decision here.

Maybe do it as an additional commit to this PR so we can easily peel it back if the folks more familiar with the rust code and direction of this library feel differently.

I think we are still in the early days of this library so it is reasonable to iterate like this.

Reply:
I would like to request that the number of worker threads be added as a field to `Compression::Zstd`, for example:

```rust

WriteOptions::new().compression(Compression::Zstd {num_worker_threads: 2})
```